GitVersionCore/BuildServers/GitHelper.cs

```diff
@@ -203,6 +205,9 @@ static void CreateMissingLocalBranchesFromRemoteTrackingOnes(Repository repo, st
                 {
                     repo.Refs.Add(localCanonicalName, new ObjectId(symbolicReference.ResolveToDirectReference().TargetIdentifier), true);
                 }
+
+                var branch = repo.Branches[branchName];
+                repo.Branches.Update(branch, b => b.TrackedBranch = remoteTrackingReferenceName);
```

Comment:
I think these statements are only executed for newly created branches, otherwise they are skipped due to the if statement in [line 192](#discussion-diff-32599197R192).